src/player.cpp

```diff
@@ -3869,7 +3869,7 @@ void player::check_needs_extremes()
                 } else if( get_kcal_percent() < 0.5f ) {
                     add_msg_if_player( m_warning, _( "You feel like you haven't eaten in days..." ) );
```

Comment:
This line also needs fixing, since the player may have just eaten while still low on calories.

Maybe we can return different messages depending on the fullness of the player, for example `"Your stomach feels so empty"` when the player is actually empty, and "`You feel weak due to malnutrition`" otherwise.